gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java

```diff
@@ -5,26 +5,41 @@
 package com.rln.gui.backend.implementation.balanceManagement.data;
 
 import com.rln.damlCodegen.model.balance.Balance;
-import java.util.Objects;
 import lombok.AllArgsConstructor;
 import lombok.Data;
 
+import java.util.Objects;
+
 @Data
 @AllArgsConstructor
 public class AccountInfo {
 
-  private final String ownerName;
-  private String providerParty;
-  private String ownerParty;
-  private String iban;
-  private String assetCode;
+    private final String ownerName;
+    private String providerParty;
+    private String ownerParty;
+    private String iban;
+    private String assetCode;
+
+    public AccountInfo(Balance balance) {
+        providerParty = balance.provider;
+        iban = balance.iban;
+        assetCode = balance.currency;
+        Objects.requireNonNull(balance.owner);
+        ownerParty = balance.owner.party.orElse(null);
+        ownerName = balance.owner.name;
+    }
 
-  public AccountInfo(Balance balance) {
-    providerParty = balance.provider;
-    iban = balance.iban;
-    assetCode = balance.currency;
-    Objects.requireNonNull(balance.owner);
-    ownerParty = balance.owner.party.orElse(null);
-    ownerName = balance.owner.name;
-  }
+    /**
+     * P   O  = A
```

Comment:
As soon as you are the provider it should be a liability .. it's maybe a bit of splitting hairs in this case, but I would say this should be a liability.

Reply:
In this case I was thinking about the "treasury" account where a bank holds an account provided by itself...

Reply:
yes - I think we want to call this a liability

Reply:
ok